internal/workflow/processing_expectations_test.go

```diff
@@ -0,0 +1,501 @@
+	"github.com/artefactual-sdps/temporal-activities/xmlvalidate"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/mock"
+	"go.artefactual.dev/tools/ref"
+
+	"github.com/artefactual-sdps/enduro/internal/a3m"
+	"github.com/artefactual-sdps/enduro/internal/datatypes"
+	"github.com/artefactual-sdps/enduro/internal/enums"
+	"github.com/artefactual-sdps/enduro/internal/workflow/activities"
+)
+
+const (
+	tempPath         = "/tmp/enduro123456"
+	extractPath      = "/tmp/enduro123456/extract"
+	transferPath     = "/home/a3m/.local/share/a3m/share/enduro2985726865"
+	prepSharedPath   = "/home/enduro/preprocessing/"
+	prepDownloadPath = "/home/enduro/preprocessing/enduro123456"
+	prepExtractPath  = "/home/enduro/preprocessing/enduro123456/extract"
+	failedSIPKey     = "Failed_SIP_name-e2ace0da-8697-453d-9ea1-4c9b62309e54.zip"
+	failedPIPKey     = "Failed_PIP_name-e2ace0da-8697-453d-9ea1-4c9b62309e54.zip"
+	sipID            = 1
+	workflowID       = 1
+	copySIPTaskID    = 100
+	valBagTaskID     = 101
+	valPREMISTaskID  = 102
+	uploadTaskID     = 103
+	reviewAIPTaskID  = 104
+	moveAIPTaskID    = 105
+	sipName          = "name.zip"
+	key              = "transfer.zip"
+	watcherName      = "watcher"
+)
+
+var (
+	ctx             = mock.AnythingOfType("*context.valueCtx")
+	sessionCtx      = mock.AnythingOfType("*context.timerCtx")
+	internalCtx     = mock.AnythingOfType("*internal.valueCtx")
+	sipUUID         = uuid.MustParse("e2ace0da-8697-453d-9ea1-4c9b62309e54")
+	locationID      = uuid.MustParse("f2cc963f-c14d-4eaa-b950-bd207189a1f1")
+	amssLocationID  = uuid.MustParse("e0ed8b2a-8ae2-4546-b5d8-f0090919df04")
+	transferID      = uuid.MustParse("65233405-771e-4f7e-b2d9-b08439570ba2")
+	aipUUID         = uuid.MustParse("9e8161cc-2815-4d6f-8a75-f003c41b257b")
+	workflowUUID    = uuid.MustParse("8fdfaea1-06ed-4cf6-8bdf-d15d80420f35")
+	startTime       = time.Date(2024, 7, 9, 16, 55, 13, 50, time.UTC)
+	retentionPeriod = ref.New(time.Second)
+)
+
+// expectationParams holds configurable parameters for setting up activity expectations.
+// Fields with default values use sensible defaults when not set, while required fields
+// must be provided for expectations that use them.
+type expectationParams struct {
+	// SIP type used for classification.
+	// Used in "classifySIP" expectation.
+	// Default: enums.SIPTypeUnknown.
+	sipType enums.SIPType
+
+	// SIP status for status updates.
+	// Used in "setStatus" and "updateSIPFailed" expectations.
+	// Required.
+	sipStatus enums.SIPStatus
+
+	// Workflow type for workflow creation.
+	// Used in "createWorkflow" expectation.
+	// Default: enums.WorkflowTypeCreateAip.
+	workflowType enums.WorkflowType
+
+	// Workflow status for workflow status updates.
+	// Used in "setWorkflowStatus" expectation.
+	// Default: enums.WorkflowStatusPending.
+	workflowStatus enums.WorkflowStatus
+
+	// Task identifier for task operations.
+	// Used in "createTask" and "completeTask" expectations.
+	// Required.
+	taskID int
```

Comment:
Having a parameter that is required but only for a couple of expectations seems like it could lead to errors and be tricky to maintain (e.g. changing which expectations use the param will require updates in `preprocessing_test.go` and the comment here).  I think it would be better to use a constructor of some sort that sets a default value for `taskID` and other params where the "zero" value doesn't make sense as a default.

Reply:
This particular field could even be removed and use always 0 as the task ID because it's used as return param and then as the input, but this is closer to reality where all tasks would get a different ID. But I'm not sure I understand your suggestion, each call to an expectation uses a new params instance, where would you add a constructor?

> (e.g. changing which expectations use the param will require updates in preprocessing_test.go and the comment here)

I think this will happen more than once and I don't see an issue with it. We'll probably add more expectations and parameters at some point.

Reply:
Because the `expectationParams` is a single struct I would expect to instantiate it once with all of the parameters needed for a test, and then reuse it for each activity.  If the parameters are meant to be specific to an activity, then I would expect each activity to have it's own parameters struct.

Reply:
What would you prefer?

Reply:
I'd prefer it being a single struct (as now) but having a `defaultParams()` constructor that returns a new instance with defaults for all the fields.

Reply:
Thanks @djjuhasz! I like the idea, let me know how it looks to you.